frontend/src/Components/Basket.js

```diff
@@ -4,57 +4,64 @@ import { useNavigate } from "react-router-dom";
 import ProductList from "./ProductList";
 import BasketItem from "./BasketItem";
 import ShopService from "../Services/ShopService";
+import PaymentForm from "./PaymentForm";
 
 const DELIVERY_FEE = 1;
 
-export default function Basket() {
+export default function Basket(props) {
+  let orderData = JSON.stringify(props.order);
+
   const navigate = useNavigate();
   const [product, setProduct] = useState([]);
-  const [basket, setBasket] = useState([])
+  const [basket, setBasket] = useState([]);
+  const [paying, setPaying] = useState(false);
+  const [order, setOrder] = useState({});
+  const [cardPayment, setCardPayment] = useState(false);
 
   useEffect(() => {
-    if (localStorage.getItem('basket') !== null) {
-        setBasket(JSON.parse(localStorage.getItem("basket")))
+    if (localStorage.getItem("basket") !== null) {
+      setBasket(JSON.parse(localStorage.getItem("basket")));
     }
-  }, [])
-
-  // let basketItems = [];
-
-  // if (localStorage.getItem("basket") !== null) {
-  //   let itemCount = JSON.parse(localStorage.getItem("basket")).length;
-
-  //   for (let i = 0; i < itemCount; i++) {
-  //     let currentItem = JSON.parse(localStorage.getItem("basket"))[i];
-  //     basketItems.push(<BasketItem product={currentItem} index={i} />);
-  //     total += currentItem.price * currentItem.quantity;
-  //   }
-  // }
+  }, []);
 
-  const placeOrder = async () => {
-    //const json = localStorage.getItem('basket')
-    let id = localStorage.getItem("userId");
-    let items = localStorage.getItem("basket");
+  const ORDER_REST_API_URL =
+    "http://localhost:5000/orders/" + localStorage.getItem("userId");
 
-    console.log(total)
-
-    const apiUrl = "http://localhost:5000/orders/" + id;
-
-    let order = {
-      totalPrice: total,
-      userID: id,
-      items: items,
+    const placeOrder = async () => {
+      //const json = localStorage.getItem('basket')
+      let id = localStorage.getItem("userId");
+      let items = localStorage.getItem("basket");
+  
+      console.log(total);
+  
+      setOrder({
+        totalPrice: total,
+        userID: id,
+        items: items,
+        type: "Cash"
+      });
+  
+      if (localStorage.getItem("userOrder") === "true") {
+        alert(
+          "You have already placed an order. Please cancel it before making a new one."
+        );
+        return;
```

Comment:
You can combine these - it's perfectly valid to return the result of calling a function:

```suggestion
        return alert(
          "You have already placed an order. Please cancel it before making a new one."
        );
```